Dockerfile.test.gpu

```diff
@@ -52,28 +54,33 @@ RUN CUDNN_MAJOR=$(cut -d '.' -f 1 <<< "${CUDNN_VERSION}"); \
         build-essential \
         g++-${GPP_VERSION} \
         moreutils \
-        libcudnn${CUDNN_MAJOR}=${CUDNN_VERSION} \
-        libnccl2=${NCCL_VERSION_OVERRIDE} \
-        libnccl-dev=${NCCL_VERSION_OVERRIDE}
+        libcudnn${CUDNN_MAJOR}=${CUDNN_VERSION} && \
+    rm -rf /var/lib/apt/lists/*
 
 # setup ssh service
 RUN ssh-keygen -f /root/.ssh/id_rsa -q -N ''
 RUN cp -v /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys
 
-# Install Python.
-RUN apt-get update -qq && apt-get install -y python${PYTHON_VERSION} python${PYTHON_VERSION}-dev python${PYTHON_VERSION}-distutils
-RUN ln -s -f /usr/bin/python${PYTHON_VERSION} /usr/bin/python
-RUN ln -s -f /usr/bin/python${PYTHON_VERSION} /usr/bin/python${PYTHON_VERSION/%.*/}
-RUN wget --progress=dot:mega https://bootstrap.pypa.io/get-pip.py && python get-pip.py && rm get-pip.py
+RUN wget -q https://repo.continuum.io/miniconda/Miniconda3-py38_4.12.0-Linux-x86_64.sh -O miniconda.sh && \
```

Comment:
The `NVTabularDataModule` does not contain any GPU-specific code, it simply calls into `nvtabular.loader.tensorflow.KerasSequenceLoader`, which I presume is tested with GPUs thoroughly. I do not see the benefits to test `NVTabularDataModule` with GPU support, especially given this has an impact on the entire Python environment.

Reply:
Tried to just `pip install nvtabular` into the original `Dockerfile.test.gpu`, but that fails to install other dependencies that require conda, e.g. `cudf`.  So, if I understand correctly, we can just leave the original `Dockerfile.test.gpu` (without `nvtabular`), which would skip the newly added `NVTabularDataModule` tests, but would still test the `PetastormDataModule` and related infrastructure.  For users interested in `NVTabular`, we can always park a separate dockerfile/image somewhere else outside of the horovod build pipeline (maybe something like `horovod/docker/horovod-nvtabular`?).

If you're OK with that, I can revert the edits to both dockerfiles (since they'd be largely unrelated to this PR then, and can always create a separate PR for the image slimming bits if desired).

Reply:
Firstly, given Horovod with NVTabular on GPU requires Conda, this should be noted in the docs, maybe in a new section between Spark and Controllers (at the level of Spark).

Secondly, I would prefer the `horovod/docker/horovod-nvtabular` image that simplifies building a GPU NVTabular environment for Horovod. Then, `nvtabular` and `miniconda` can be kept out of `Dockerfile.test.gpu`, but the size improvements (`rm -rf /var/lib/apt/lists/*`) should stay. We can add the image to the CI so we always know this image builds:
https://github.com/horovod/horovod/blob/32b5d3ecbb419bcebedf1a456f713ef0c05b578a/.github/gen-workflow-ci.py#L817

Reply:
OK, updated `docs/spark.rst` with some comments about conda/docker w.r.t. `NVTabularDataModule`, since currently the only mention and it's fairly specific to the `KerasEstimator` at the moment.  Also, moved the NVTabular dockerfile into `horovod/docker/horovod-nvtabular` and removed `nvtabular` and `conda` from the `Dockerfile.test.*` files.